tests/gocase/unit/quit_test.go

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package quit
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/apache/incubator-kvrocks/tests/gocase/util"
+	"github.com/go-redis/redis/v9"
+	"github.com/stretchr/testify/require"
+)
+
+func reconnect(srv *util.KvrocksServer, rdb **redis.Client) error {
+	if err := (*rdb).Close(); err != nil {
+		return err
+	}
+	*rdb = srv.NewClient()
+	return nil
+}
+
+func TestPipeQuit(t *testing.T) {
+	srv := util.StartServer(t, map[string]string{})
+	defer srv.Close()
+	rdb := srv.NewClient()
+	ctx := context.Background()
+
+	t.Run("QUIT returns OK", func(t *testing.T) {
+		require.NoError(t, reconnect(srv, &rdb))
+		pipe := rdb.Pipeline()
+		cmd1 := pipe.Do(ctx, "quit")
+		cmd2 := pipe.Do(ctx, "ping")
+		_, err := pipe.Exec(ctx)
+		require.Equal(t, "OK", cmd1.Val())
+		require.Equal(t, nil, cmd2.Val())
+		require.Error(t, err)
+	})
```

Comment:
I think we do not need to use pipeline in this test?

Here is the corresponding TCL code:
```
    test "QUIT returns OK" {
        reconnect
        assert_equal OK [r quit]
        assert_error * {r ping}
    }
```

Reply:
```go
cmd1 := rdb.Do(ctx, "quit")
cmd2 := rdb.Ping(ctx)
require.Equal(t, "OK", cmd1.Val())
require.Equal(t, nil, cmd2.Val())
```

because go-redis don't support `quit`.And i find use `rdb.Do` to excute `quit` at here is also not work. 

like above,`ping` should return `nil`, but returns `pong` when actually running.

Reply:
@IoCing Can see this example: https://github.com/apache/incubator-kvrocks/blob/unstable/tests/gocase/unit/keyspace/keyspace_test.go#L106

Reply:
If you really want to use pipeline here, I think the identical code will be:
```
pipe := rdb.Pipeline()
cmd1 := pipe.Do(ctx, "quit")
_, err := pipe.Exec(ctx)
// process cmd1
pipe = rdb.Pipeline()
cmd2 := pipe.Do(ctx, "ping")
_, err = pipe.Exec(ctx)
```

But in your code:
```
pipe := rdb.Pipeline()
cmd1 := pipe.Do(ctx, "quit")
cmd2 := pipe.Do(ctx, "ping")
_, err := pipe.Exec(ctx)
```

these two command is packed into one pipeline, which is not identical to the TCL code.

Reply:
ok，i revise it now

Reply:
```go
require.NoError(t, reconnect(srv, &rdb))
pipe := rdb.Pipeline()
cmd1 := pipe.Do(ctx, "quit")
_, err := pipe.Exec(ctx)
require.Equal(t, "OK", cmd1.Val())
require.NoError(t, err)
// process cmd1
pipe = rdb.Pipeline()
cmd2 := pipe.Do(ctx, "ping")
_, err = pipe.Exec(ctx)
require.Equal(t, nil, cmd2.Val())
require.Error(t, err)
```

This method is not work.

![image-20220917223238253](https://iocing-image.oss-cn-shenzhen.aliyuncs.com/img/image-20220917223238253.png)

 i think i can use tcpclient like what I first implemented.

```go
trdb := srv.NewTCPClient()
require.NoError(t, trdb.Write("*1\r\n$4\r\nquit\r\n"))
resQuit, errQuit := trdb.ReadLine()
require.Equal(t, "+OK", resQuit)
require.NoError(t, errQuit)
require.NoError(t, trdb.Write("*1\r\n$4\r\nping\r\n"))
resPing, errPing := trdb.ReadLine()
require.Equal(t, "", resPing)
require.Error(t, errPing)
```